File: algorithm/programmers/level_01/완주하지_못한_선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String, Integer> result = new HashMap<String, Integer>();
        
        for (String name : participant) {
            result.put(name, result.getOrDefault(name, 0) + 1);
        }
        
        for (String name : completion) {
            result.put(name, result.get(name) - 1);
        }
        
        for (Map.Entry<String, Integer> entry : result.entrySet()) {
            if (entry.getValue() == 1) {
                answer = entry.getKey();
                break;
            }
        }
        
        return answer;
    }
}

File: algorithm/programmers/level_01/완주하지_못한_선수.py
from collections import defaultdict


def solution(participant, completion):
    answer = ''
    dictionary = defaultdict(int)

    for name in participant:
        dictionary[name] += 1

    for name in completion:
        dictionary[name] -= 1

    for key, value in dictionary.items():
        if value == 1:
            answer = key
            break

    return answer
